ChatApp/Presentation/View/SwitchView/SwitchModeView.swift

```diff
@@ -0,0 +1,38 @@
+import UIKit
+
+// MARK: Switch Protocol
+protocol SwitchModeViewDelegate: AnyObject {
+    func switchModeView(with state: SwitchModeView.ButtonState)
+}
+
+final class SwitchModeView: UIButton {
+    
+    weak var delegate: SwitchModeViewDelegate?
+    
+    // MARK: Init
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setUpConstraint()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    // MARK: Constraint
+    private func setUpConstraint() {
```

Comment:
does not look much like setting up a constraint to me. In fact translatesAutoresizingMaskIntoConstraints should be performed from the outside where this button is instantiated and constrained